src/stdlib_linalg.fypp

```diff
@@ -5,13 +5,21 @@ module stdlib_linalg
   !! ([Specification](../page/specs/stdlib_linalg.html))
   use stdlib_kinds, only: sp, dp, qp, &
     int8, int16, int32, int64
+  use stdlib_error, only: error_stop
   implicit none
   private
 
   public :: diag
   public :: eye
   public :: trace
   public :: outer_product
+  public :: is_square
+  public :: is_diagonal
+  public :: is_symmetric
+  public :: is_skew_symmetric
+  public :: is_hermitian
+  public :: is_triangular
+  public :: is_hessenberg
```

Comment:
Would it be wise to move the implementations of all these functions fo submodules?

Reply:
I had considered that, but it seemed like it would further pollute the already cluttered `src` directory.
As long as this is fine with people, I can move them into their own submodules.

Reply:
I don't have time to find the thread, but if I recall correctly our plan was that once a module (say `linalg`) reaches a certain size, we would introduce a new folder like `linalg/` in the `src/` directory. We haven't done it yet for the `stats` routines which would be another candidate. @jvdp1, what do you think?

Reply:
A separate folder is actually what I wanted to ask for, but didn't know the overall plan or want to complicate things.
I am very much on board with having separate folders for modules.

Reply:
> I don't have time to find the thread, but if I recall correctly our plan was that once a module (say `linalg`) reaches a certain size, we would introduce a new folder like `linalg/` in the `src/` directory. We haven't done it yet for the `stats` routines which would be another candidate. @jvdp1, what do you think?

I don't recall this discussion/thread (but it doesn't mean I didn't read it or didn't participate to it ;) ). However, I think it would be a good idea of having separate folders for modules/a set of modules related to the same topic.

For now, I would suggest to keep it as it is, or to only create submodules. We can discuss in an issue how to create folders, and if it is appropriate (also by keeping in mind that it must be compatible with fpm).

Reply:
There is one issue to keep in mind with FORD when using subdirectories for the Fortran files, because only the basename will be visible in the documentation, like here: https://toml-f.github.io/toml-f/lists/files.html

Reply:
Looking back, it was a single comment from Zaak Beekman in https://github.com/fortran-lang/stdlib/issues/7#issuecomment-566848401:
> As stuff gets added if src gets too cluttered you can break
out more nesting and structure. E.g. `src/strings/ascii.f90` etc. where
other string stuff gets gets grouped there.

I agree we postpone the folder changes for a new PR, which will be linked with a newly created issue.